### PR TITLE
RHOAIENG-371: Add pipeline support for git private repo using an authentication token

### PR DESCRIPTION
**Pull request by jackdelahunt**
## Description

**This is blocked by https://github.com/opendatahub-io/ai-edge/pull/284**

Users can now fetch from private repos, this pr adds that ability to the aiedge-e2e pipeline but once #284 is merged these changes will be moved there. And with that any docs that are needed to give users info on the new feature.

The e2e tests have beed updated to support this optional feature and have been passing when using the `git_fetch` test when cloning from my own private repo.

## How Has This Been Tested?
- Using e2e tests fill in `token` and `username` in `config.json`
- All tests should pass

I have a private example repo if needed for testing confirmation

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by MarianMacik** [comment]
This now has conflicts after #284 was merged.

**Comment by jackdelahunt** [comment]
@MarianMacik fixed now

**Comment by biswassri** [comment]
/hold

@jackdelahunt Putting this on-hold as some part of the git-auth implementation needs to be updates based on #290 and https://tekton.dev/docs/pipelines/git-resolver/

**Comment by adelton** [comment]
Hello @jackdelahunt, could you please rebase so that we have a clean CI and we can look at the changes in the context of the latest repo content?

**Comment by jackdelahunt** [comment]
@adelton done now

File: test/e2e-tests/support/setup.go
Comment (jackdelahunt) [reply test/e2e-tests/support/setup.go:131]: I don't know how I mixed that up, thanks

File: examples/tekton/aiedge-e2e/templates/credentials-git.secret.yaml.template
Comment (jackdelahunt) [reply examples/tekton/aiedge-e2e/templates/credentials-git.secret.yaml.template:10]: Removed, the token field still needs to be there for the .git-credentials but you are right the `token` is not needed, I wasnt 100% on this

File: manifests/pipelines/git-fetch-pipeline.yaml
Comment (jackdelahunt) [reply manifests/pipelines/git-fetch-pipeline.yaml:361]: I am not 100% sure I understand what you mean. 

Are you saying that from a users perspective you can add you git credentials like so (or something close):
```yaml
"https://{username}:{github_pat_token}@github.com
https://{username}:{gitlab_pat_token}@gitlab.com
"
```

But for the e2e tests we just use one set of a username/token because we are enforcing they are from the same domain?

Comment (MarianMacik) [reply manifests/pipelines/git-fetch-pipeline.yaml]: @jackdelahunt yes, exactly.
1. We can have just one secret for many repositories.
2. We will anyway use just one par of user/token credentials to simplify our test setup, i.e. it is not needed to be able to set separate user/token for Containerfile Git repo and for Model Git repo in our use case. That will simplify the test setup, i.e. no need for building a more complex string with more than one username/token URL etc.

File: test/e2e-tests/support/setup.go
Comment (MarianMacik) [inline test/e2e-tests/support/setup.go:59]:
```suggestion
	// needed as a workspace for the pipelines
```

File: test/e2e-tests/README.md
Comment (MarianMacik) [reply test/e2e-tests/README.md:37]: Please also fix the readme as the parameters have been moved.

**Review by MarianMacik — COMMENTED** [review]
Looks good now, just correction of the docs is needed.

**Review by MarianMacik — APPROVED** [review]
Looks good now, thank you!

/approve

**Pull request merged**
